binary_search_tree/BinarySearchTree.py

```diff
@@ -0,0 +1,334 @@
+class Node():
+    # This class holds an actual Node in our BST
+
+    # the actual data which is stored in the node
+    __data = None
+    # left child of the node
+    __left = None
+    # right child of the node
+    __right = None
+
+    def __init__(self, data=None, left=None, right=None):
+        # here self.data, self.left, self.right are not variables
+        # these are setters which are defined below
+        self.data = data
+        self.left = left
+        self.right = right
+
+    @property
+    def data(self):
+        # adding @property makes this a getter and can be accessed by ObjectName.data
+
+        return self.__data
+
+    @data.setter
+    def data(self, value):
+        # we need setter for "__data" since its a private variable
+        # Use of setters is to validate data or perform some action before assiging the data
+        # we only want our data to be a number/int
+        if isinstance(value, int):
+            self.__data = value
+        else:
+            self.__data = None
+            print("Invalid data: {} only integers(int) are accepted.".format(value))
+
+    # now we will repeat the same for "left" and "right" variables too
+    @property
+    def left(self):
+        return self.__left
+
+    @left.setter
+    def left(self, value):
+        # since left is not just any variable but a node itself
+        if (isinstance(value, Node)) or (value is None):
+            self.__left = value
+        else:
+            print("Invalid left node, should be an instance of class 'Node'.")
+
+    # repeat the same for right node
+    @property
+    def right(self):
+        return self.__right
+
+    @right.setter
+    def right(self, value):
+        # since right is not just any variable but a node itself
+        if (isinstance(value, Node)) or (value is None):
+            self.__right = value
+        else:
+            print("Invalid right node, should be an instance of class 'Node'.")
+
+    def insert(self, value):
+        # no duplicate data
+        if self.data == value:
+            print("Value: {} already exists.".format(value))
+            return False
+        elif self.data > value:
+            # if there is a left child insert data there
+            if self.left:
+                return self.left.insert(value)
+            # else create a new node
+            else:
+                self.left = Node(value)
+                return True
+        # repeating same steps for the right child
+        else:
+            if self.right:
+                return self.right.insert(value)
+            else:
+                self.right = Node(value)
+                return True
+
+    def find(self, value):
+        if self.data == value:
+            print("Value: {} FOUND!".format(value))
+            return True
+        elif self.data > value:
+            if self.left:
+                return self.left.find(value)
+            else:
+                print("Value: {} NOT FOUND!".format(value))
```

Comment:
Same here, Rather than printing the error message why not raise an exception?

Reply:
Since this is a find function, we can skip exceptions as this would just return `True` and `False.